pvlib/irradiance.py

```diff
@@ -1513,11 +1513,14 @@ def poa_error(ghi):
                         full_output=True,
                         disp=False,
                         )
-    except ValueError:
-        # this occurs when poa_error has the same sign at both end points
-        ghi = np.nan
-        conv = False
-        niter = -1
+    except ValueError as e:
+        # this may occur because poa_error has the same sign at both end points
+        if "f(a) and f(b) must have different signs" in e.args:
```

Comment:
Minor concern - what if scipy changes the text of the error message? I think I prefer to simply `raise e` and exit, and change the "this occurs" comment to "occurs when .... both end points, for example"

Reply:
From reading the code I have understood that `f(a) and f(b) must have different signs` ValueErrors do need to be skipped. It probably was that @adriesse coded it to take into account some cases where the set of arguments passed onto bisect didn't make sense for it, but were generated from a reasonable input. (@adriesse can you confirm my suspicion?)

`scipy.optimize.bisect` also raises `ValueError` for other reasons, like a negative `xtol`. `ghi_from_poa_adriesse_2023` does not allow other problematic combinations in `bisect` due to the limited number of arguments you can pass onto it.

I'm testing the `xtol` argument propagates correctly by passing a nonsensical value (negative). In that case, a `ValueError` exception should be risen. This matching allows for raising that kind of exception from scipy. The opposite match can also be used (`if "xtol too small " in e.args: ...`).

I think letting the exception be risen from the scipy side hints any user to read the bisect documentation. That documentation is better in terms of what `xtol` is.

> what if scipy changes the text of the error message? 

I can change the condition to be `"different signs" in e.args` to preserve behaviour and be laxer or do the `xtol` validation internally in PVLIB. In any case, I doubt a numerical method like bisect will change the cause message in a failure like that.

Reply:
> do the xtol validation internally in PVLIB.

I think this is a much better solution.